gui/src-tauri/src/main.rs

```diff
@@ -68,88 +69,128 @@ fn main() -> Result<()> {
+		dir.join("Library/Logs")
+			.join(&context.config().identifier)
+	});
+
+	#[cfg(not(target_os = "macos"))]
+	let path = dirs_next::data_dir()
+		.ok_or(Error::UnknownPath)
+		.map(|dir| dir.join(&context.config().identifier).join("logs"));
+
+	Ok(Logger::try_with_env_or_str("info")?
+		.log_to_file(FileSpec::default().directory(path.expect("We need a log dir")))
+		.format_for_files(|w, now, record| util::logger_format(w, now, record, false))
+		.format_for_stderr(|w, now, record| util::logger_format(w, now, record, true))
+		.rotate(
+			Criterion::Age(Age::Day),
+			Naming::Timestamps,
+			Cleanup::KeepLogFiles(2),
+		)
+		.duplicate_to_stderr(Duplicate::All)
+		.write_mode(WriteMode::BufferAndFlush)
+		.start()?)
+}
+#[cfg(windows)]
+fn setup_webview2() -> Result<()> {
+	use crate::util::webview2_exists;
+	use win32job::{ExtendedLimitInfo, Job};
+
+	let mut info = ExtendedLimitInfo::new();
+	info.limit_kill_on_job_close();
+	let job = Job::create_with_limit_info(&mut info).expect("Failed to create Job");
+	job.assign_current_process()
+		.expect("Failed to assign current process to Job");
+
+	// We don't do anything with the job anymore, but we shouldn't drop it because that would
+	// terminate our process tree. So we intentionally leak it instead.
+	std::mem::forget(job);
+
+	if !webview2_exists() {
+		// This makes a dialog appear which let's you press Ok or Cancel
+		// If you press Ok it will open the SlimeVR installer documentation
+		use rfd::{MessageButtons, MessageDialog, MessageDialogResult, MessageLevel};
+
+		let confirm = MessageDialog::new()
+			.set_title("SlimeVR")
+			.set_description("Couldn't find WebView2 installed. You can install it with the SlimeVR installer")
+			.set_buttons(MessageButtons::OkCancel)
+			.set_level(MessageLevel::Error)
+			.show();
+		if confirm == MessageDialogResult::Ok {
+			open::that("https://docs.slimevr.dev/server-setup/installing-and-connecting.html#install-the-latest-slimevr-installer").unwrap();
+		}
+		return Ok(());
+	}
+	return Ok(());
+}
+
+fn check_environment_variables() {
+	let java_options = var("_JAVA_OPTIONS").ok();
+	let java_tool_options = var("JAVA_TOOL_OPTIONS").ok();
+
+	if java_options.is_some() || java_tool_options.is_some() {
+		let mut env_warn_msg = String::new();
+
+		if let Some(env) = java_options {
+			string_append_list_style(&mut env_warn_msg, "_JAVA_OPTIONS");
+			log::warn!("_JAVA_OPTIONS is set: {}", env);
+		}
+
+		if let Some(env) = java_tool_options {
+			string_append_list_style(&mut env_warn_msg, "JAVA_TOOL_OPTIONS");
+			log::warn!("JAVA_TOOL_OPTIONS is set: {}", env);
+		}
+
+		// Display the warning dialog
+		rfd::MessageDialog::new()
+            .set_title("SlimeVR")
+            .set_description(&format!("You have environment variables {} set, which may cause the SlimeVR Server to fail to launch properly.", env_warn_msg))
+            .set_level(rfd::MessageLevel::Warning)
+            .show();
+	}
```

Comment:
not a big fan of this, maybe via filter map it would look nicer? let's try

```suggestion
	const ENVS_TO_CHECK: &[&str] = &["_JAVA_OPTIONS", "JAVA_TOOL_OPTIONS"]; 
	let checked_envs = ENVS_TO_CHECK.filter_map(|e| {
		let Some(data) = var(e) else { return None; };
		log::warn!("{e} is set to: {data}");
		Some(e)
	}).join(", ");

	if !checked_envs.is_empty() {
		rfd::MessageDialog::new()
			.set_title("SlimeVR")
			.set_description(&format!("You have environment variables {checked_envs} set, which may cause the SlimeVR Server to fail to launch properly."))
			.set_level(rfd::MessageLevel::Warning)
			.show();
	}
```

Reply:
That code doesn't even compile @w@;

Reply:
Well first I get

```
no method named `filter_map` found for reference `&'static [&'static str]` in the current scope
`&'static [&'static str]` is not an iterator
```
Which adding .into_iter() `ENVS_TO_CHECK.into_iter().filter_map` fixes

Then `Some(data)` gives me this:

```
mismatched types
expected enum `Result<std::string::String, VarError>`
   found enum `std::option::Option<_>`
```

Then `.join` gives me this:

```
no method named `join` found for struct `FilterMap` in the current scope
method not found in `FilterMap<Iter<'_, &str>, {closure@main.rs:167:58}>`
```

Reply:
yeah sadly the join part requires itertools, i forgor, well pull it as a dependency, its pretty useful for iterating

Reply:
and this is with the other errors fixed
```rs
	const ENVS_TO_CHECK: &[&str] = &["_JAVA_OPTIONS", "JAVA_TOOL_OPTIONS"];
	let checked_envs = ENVS_TO_CHECK.into_iter().filter_map(|e| {
		let Ok(data) = env::var(e) else { return None; };
		log::warn!("{e} is set to: {data}");
		Some(e)
	}).join(", ");
	if !checked_envs.is_empty() {
		rfd::MessageDialog::new()
			.set_title("SlimeVR")
			.set_description(&format!("You have environment variables {checked_envs} set, which may cause the SlimeVR Server to fail to launch properly."))
			.set_level(rfd::MessageLevel::Warning)
			.show();
	}
```

Reply:
> yeah sadly the join part requires itertools, i forgor, well pull it as a dependency, its pretty useful for iterating

That moment when rust is... *Terrible joke here* Illiterate.